client/src/EntryForm.js

```diff
@@ -15,12 +15,62 @@ export default function EntryForm({ entry, onSubmit }) {
   function handleSubmit(event) {
     event.preventDefault();
     const newEntry = { title, photoUrl, notes };
-    if (entry) {
-      updateEntry({ ...entry, ...newEntry });
-    } else {
-      addEntry(newEntry);
+    try {
+      if (entry) {
+        addUpdateEntry(entry.entryId, newEntry);
+      } else {
+        addEntry(newEntry);
+      }
+      onSubmit();
+    } catch (err) {
+      console.error(err);
     }
-    onSubmit();
+  }
+
+  async function addEntry(newEntry) {
+    const newTask = await createEntry(newEntry);
+    setEntries((prev) => prev.concat([newTask]));
+  }
+
+  async function createEntry(entry) {
+    const req = {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(entry),
+    };
+    const res = await fetch('/api/journal', req);
+    if (!res.ok) throw new Error(`fetch Error ${res.status}`);
+    return await res.json();
+  }
+
+  async function addUpdateEntry(entryId, newEntry) {
```

Comment:
Why `addUpdateEntry` instead of just `updateEntry`? Oh, it does more than just call the backend. I see, there's a name conflict with `updateEntry` on line 56. Naming is hard!